### DISCUSSION TITLE
[FR] New relational field type: section field

### DISCUSSION THREAD

**Discussion by thupsi** [craftcms/cms#9736] [Ideas] [answered]
When it comes to **content modelling** and **author experience** in Craft, one thing that I keep getting hit with is a lack of a **sections field,** like the relations fields for entry, assets, categories etc.

I think that could be very useful in a variety of cases. Example: I may have a section and I want to have an index page for that (pretty common scenario). I can define this relationship in various ways, but none of them is as flexible, dynamic and easily managed as a relational field.

A relevant example of such functionality, can be seen in the concept of "[mounting](https://statamic.dev/collections#mounting)"  in Statamic, where you basically assign a section to a given page. In Craft this could be even more powerful, because with a relational section field you could corelate a section with any number of pages, so, if needed, you could have many different indexes.

Bonus: I think that it wouldn't be difficult to implement in Craft. The UI is there, and the logic is mostly there.

I would appreciate any comments, especially from the Craft team!

**Comment by brandonkelly** [comment, marked as answer]
You can roll your own using [LJ Dynamic Fields](https://plugins.craftcms.com/craft-dynamic-fields) – they even show that as one of the examples.

**Reply by thupsi** [reply]
Thanks @brandonkelly! That indeed is a solution.

I still believe a field like this (section field, not dynamic field) should make it to the core, because of two reasons:
- Required or useful in a great variety of cases
- Easy to implement, no added bloat (I think)

In any case, thanks for you answer!

**Reply by brandonkelly** [reply]
Yeah not ruling it out, but wanted to provide a way of doing it in the meantime.
